server/src/main/java/org/opensearch/index/IndexModule.java

```diff
@@ -141,6 +143,17 @@ public final class IndexModule {
         Property.NodeScope
     );
 
+    /**
+     * Index setting which used to determine how the data is cached locally fully or partially
+     */
+    public static final Setting<LocalityType> INDEX_STORE_LOCALITY_SETTING = new Setting<>(
+        "index.store.locality",
+        LocalityType.FULL.name(),
+        LocalityType::getValueOf,
+        Property.IndexScope,
+        Property.NodeScope
```

Comment:
considering this setting would have to be updated for migrations, should it be a dynamic setting?